include/Model/Languages.php

```diff
@@ -602,22 +602,36 @@ public function get_default() {
 
 	/**
 	 * Updates the default language.
-	 * taking care to update the default category & the nav menu locations.
+	 * Takes care to update default category, nav menu locations, and flushes cache and rewrite rules.
 	 *
 	 * @since 1.8
 	 * @since 3.7 Moved from `PLL_Admin_Model::update_default_lang()` to `WP_Syntex\Polylang\Model\Languages::update_default()`.
+	 *            Returns a `WP_Error` object.
 	 *
 	 * @param string $slug New language slug.
-	 * @return void
+	 * @return WP_Error A `WP_Error` object containing possible errors during slug validation/sanitization.
 	 */
-	public function update_default( $slug ): void {
+	public function update_default( $slug ): WP_Error {
+		$prev_default_lang = $this->options->get( 'default_lang' );
+		$errors            = $this->options->set( 'default_lang', $slug );
+
+		if ( $errors->has_errors() ) {
+			return $errors;
+		}
+
+		$default_lang = $this->options->get( 'default_lang' );
+
+		if ( $prev_default_lang === $default_lang ) {
+			return new WP_Error();
+		}
```

Comment:
1. Good idea to check that the new default language is different but I would do this first.
2. Why reading the new option once it has been set. If there are no errors it can be only what has been set. Do I miss something? 
```suggestion
	public function update_default( $default_lang ): WP_Error {
		$prev_default_lang = $this->options->get( 'default_lang' );

		if ( $prev_default_lang === $default_lang ) {
			return new WP_Error();
		}

		$errors = $this->options->set( 'default_lang', $slug );

		if ( $errors->has_errors() ) {
			return $errors;
		}
```

Reply:
On both points: because the sanitized value could differ from the raw value.
Except that it is not possible in this specific case: the validation will use the regex pattern, and the sanitization will only cast the value to a string. So you're right, since the sanitization won't modify the value, it seems safe to compare the old value at the beginning.